### PR TITLE
Feat/vic/button component

### PR DESCRIPTION
**Pull request by vifar**
Adding button component

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "prettier.singleQuote": true,
   "prettier.printWidth": 100,
   "prettier.tabWidth": 2,
-  "prettier.trailingComma": "all",
```
Comment (xylocopaCaerulea) [inline .vscode/settings.json]: Why did you remove the trailing commas?

File: src/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const Button: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
+  const { children, ...remainingProps } = props;
```
Comment (xylocopaCaerulea) [inline src/Button/Button.tsx]: Why did you use the rest operator to save the remaining props into 'remainingProps' instead of destructuring them out?

File: src/Button/DefaultButton.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Button from './Button';
+import '../index';
+
+interface DefaultButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: string
```
Comment (xylocopaCaerulea) [inline src/Button/DefaultButton.tsx]: Limiting the children to just 'string' is fine for right now but we might need to revisit this later on if we decide to allow icons to be parsed down to the buttons as well.

File: src/Button/__tests__/defaultButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, cleanup } from '@testing-library/react';
+import { DefaultButton } from '../index';
+
+afterEach(cleanup);
+
+describe('Test that DefaultButton will render without crashing', () => {
```
Comment (xylocopaCaerulea) [inline src/Button/__tests__/defaultButton.test.tsx]: You don't need to test that it will render without crashing because react will test for that already.  Instead, you should test that the button is in the document or on the DOM and the different interactions the user can have with the different states of the button.

ie: Does it trigger a click if the user clicks it?  Does it allow the user to click it if it's disabled? Etc...

File: src/index.scss [hunk 5]
```diff
@@ -1,3 +1,5 @@
+@import '../node_modules/uswds/dist/scss/uswds.scss';
+
 .uswds-thing {
```
Comment (xylocopaCaerulea) [inline src/index.scss]: This class name does not seem to be super descriptive.  Was this just for testing?

File: src/Button/Button.story.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <DefaultButton className={'usa-button'}>
+      {'Click Me'}
+    </DefaultButton >
+  )
+});
+
+stories.add('Secondary Color', () => {
+  return (
+    <DefaultButton className={'usa-button usa-button--secondary'}>
```
Comment (xylocopaCaerulea) [inline src/Button/Button.story.tsx]: Since 'usa-button' is necessary for every button, you can bake this classname into the `<DefaultButton>` component and expand the classes for the other buttons.  In fact, we could define other button components with the appropriate classnames baked in so that the implementer does not need to worry about which class name is necessary to theme the buttons, they just have to focus on which button to import.

File: .vscode/settings.json (hunk 1 above)
Comment (vifar) [reply .vscode/settings.json]: @xylocopaCaerulea I removed it because it was adding unnecessary commas.

File: src/index.scss (hunk 5 above)
Comment (vifar) [reply src/index.scss]: @xylocopaCaerulea yes `.uswds-thing` was just for the initial testing, I can remove it.

File: src/Button/Button.story.tsx (hunk 6 above)
Comment (vifar) [reply src/Button/Button.story.tsx]: @xylocopaCaerulea so you're saying I should make a separate file for every class of button? I can do that, it seems cleaner as well.

File: src/Button/DefaultButton.tsx (hunk 3 above)
Comment (vifar) [reply src/Button/DefaultButton.tsx]: @xylocopaCaerulea so they type should be any?

**Comment by josxchang** [comment]
@vifar If you could build out the storybook for those components that would be wonderful so we can start seeing it show up. Storybook should be set up at this point so let me know if you run into any issues doing so.

File: src/index.tsx [hunk 7]
```diff
@@ -1,10 +1 @@
-import * as React from 'react';
-
 import './index.scss';
```
Comment (josxchang) [inline src/index.tsx]: You will need to add the export statements here to export the components that you've exported in index.ts in the Button directory. This is the root index of SRC

**Comment by vifar** [comment]
> @vifar If you could build out the storybook for those components that would be wonderful so we can start seeing it show up. Storybook should be set up at this point so let me know if you run into any issues doing so.

@josxchang I've built out the stories for all the buttons. They should be in `Button.story.tsx`. Let me know if there's something that I'm missing.

**Comment by josxchang** [comment]
> > @vifar If you could build out the storybook for those components that would be wonderful so we can start seeing it show up. Storybook should be set up at this point so let me know if you run into any issues doing so.
> 
> @josxchang I've built out the stories for all the buttons. They should be in `Button.story.tsx`. Let me know if there's something that I'm missing.

Looks good! I must have missed it. Sweet!
